RevenueCatUI/CustomerCenter/ViewModels/SubscriptionDetailViewModel.swift

```diff
@@ -69,14 +69,32 @@ final class SubscriptionDetailViewModel: BaseManageSubscriptionViewModel {
             )
         }
 
+    func didAppear() {
+        actionWrapper.onCustomerCenterPromotionalOfferSuccess { [weak self] in
+            self?.refreshPurchase()
+        }
+
+        actionWrapper.onCustomerCenterShowingManageSubscriptions { [weak self] in
+            self?.customerInfoViewModel.manageSubscriptionsSheet = true
+        }
```

Comment:
Maybe this wouldn't happen right now, but I'm afraid that, if we ever make a navigation/modal from this screen and `didAppear()` could be called again when back on this screen, we would have repeated callbacks, as the old ones are not cancelled (I think).

For that reason, I think the subscriber (in this case, this ViewModel), should be the one managing the `cancellable`, so that at the beginning of `didAppear()` we can cancel all the previous `cancellables` and re-add them

Reply:
And the subjects can (and should) still be `private` as done in this PR. But we should create a type-erased publisher for each subject that is the one used by subscribers. Something like this

```swift
private let restoreStarted = PassthroughSubject<Void, Never>()
var onRestoreStarted: AnyPublisher<Void, Never> {
    restoreStarted.eraseToAnyPublisher()
}
```

This way, subscribers would register to the `onRestoreStarted` publisher and would be able to handle cancellation by themselves.